### PR TITLE
[Core] az --version: Show command instruction and detailed instruction link when updates available

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**  

We can now rely on the `AZ_INSTALLER` env variable to know how the running Azure CLI is installed, so we can show specific instructions on how to update Azure CLI.

Example for Ubuntu:

> You have 2 updates available. Consider updating your CLI installation with 'sudo apt-get update && sudo apt-get install --only-upgrade -y azure-cli'. Detailed instructions can be found at https://aka.ms/doc/UpdateAzureCliApt

For windows, for simplicity let's just recommend the MSI link for upgrading(The upgrade cmdlet on powershell needs to be run as Administrator and I'm still looking for the alternative to [psutil](https://github.com/Azure/azure-cli/pull/11665) to decide if CLI is running in Powershell):

> You have 2 updates available. Consider updating your CLI installation with 'https://aka.ms/installazurecliwindows'. Detailed instructions can be found at https://aka.ms/doc/UpdateAzureCliMsi

For pip, 2 options for Linux and MacOS:
Linux/MacOs:

> You have 2 updates available. Consider updating your CLI installation with 'pip install --upgrade azure-cli' or 'curl -L https://aka.ms/InstallAzureCli | bash' if you used our script for installing. Detailed instructions can be found at https://aka.ms/doc/UpdateAzureCliLinux

Windows:

> You have 2 updates available. Consider updating your CLI installation with 'pip install --upgrade azure-cli'

**Testing Guide**  

**History Notes**  

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
for upgrade

File: src/azure-cli-core/azure/cli/core/__init__.py [hunk 1]
```diff
@@ -29,6 +31,16 @@
                    'content_version', 'kwargs', 'client', 'no_wait']
 EVENT_FAILED_EXTENSION_LOAD = 'MainLoader.OnFailedExtensionLoad'
 
+_PACKAGE_UPGRADE_INSTRUCTIONS = {"YUM": ("sudo yum update -y azure-cli", "https://aka.ms/doc/UpdateAzureCliYum"),
+                                 "ZYPPER": ("sudo zypper refresh && sudo zypper update -y azure-cli", "https://aka.ms/doc/UpdateAzureCliZypper"),
+                                 "DEB": ("sudo apt-get update && sudo apt-get install --only-upgrade -y azure-cli", "https://aka.ms/doc/UpdateAzureCliApt"),
+                                 "HOMEBREW": ("brew update && brew upgrade azure-cli", "https://aka.ms/doc/UpdateAzureCliHomebrew"),
+                                 "PIP": ("curl -L https://aka.ms/InstallAzureCli | bash", "https://aka.ms/doc/UpdateAzureCliLinux"),
+                                 "MSI": ("https://aka.ms/installazurecliwindows", "https://aka.ms/doc/UpdateAzureCliMsi"),
+                                 "DOCKER": ("docker pull mcr.microsoft.com/azure-cli", "https://aka.ms/doc/UpdateAzureCliDocker")}
```
Comment (yungezz) [inline src/azure-cli-core/azure/cli/core/__init__.py:40]: just confirm, those short urls are all maintained actively

File: src/azure-cli-core/azure/cli/core/__init__.py [hunk 2]
```diff
@@ -96,9 +108,33 @@ def show_version(self):
         if updates_available == -1:
             logger.warning('Unable to check if your CLI is up-to-date. Check your internet connection.')
         elif updates_available:
-            logger.warning('You have %i updates available. Consider updating your CLI installation. '
-                           'Instructions can be found at https://docs.microsoft.com/en-us/cli/azure/install-azure-cli',
-                           updates_available)
+            warning_msg = 'You have %i updates available. Consider updating your CLI installation'
+            from azure.cli.core._environment import _ENV_AZ_INSTALLER
+            installer = os.getenv(_ENV_AZ_INSTALLER)
```
Comment (yungezz) [inline src/azure-cli-core/azure/cli/core/__init__.py:113]: do we really need to print different messages based on platform? what about show all upgrade instructions?

File: src/azure-cli-core/azure/cli/core/__init__.py (hunk 1 above)
Comment (fengzhou-msft) [reply src/azure-cli-core/azure/cli/core/__init__.py:40]: all of them are active now.

File: src/azure-cli-core/azure/cli/core/__init__.py (hunk 2 above)
Comment (fengzhou-msft) [reply src/azure-cli-core/azure/cli/core/__init__.py:113]: The current [link](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) shown is doing something similiar, it puts all the links to each package in one page. But we got complains about it's not being useful, some users may not know their operating system well, they have difficulties to find the right instruction.

**Comment by jiasli** [comment]
It would be helpful to show some rendered results in the PR description.

File: src/azure-cli-core/azure/cli/core/util.py [hunk 3]
```diff
@@ -828,3 +825,22 @@ def user_confirmation(message, yes=False):
...
         raise CLIError(
             'Unable to prompt for confirmation as no tty available. Use --yes.')
+
+
+def get_linux_distro():
+    if platform.system() != 'Linux':
+        return None, None
+
+    try:
+        with open('/etc/os-release') as lines:
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/util.py:835]: Copy and paste a sample for other reviewers:

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.2 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.2 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

**Pull request merged**